Updated openapi spec to support feedback and users

sremsen opened the pull request:
**Changelog:**
- Added Mintlify feedback buttons
- Updated typescript documentation to support feedback field in the example
- Removed old Feedback schema in the Open API
- Added new createfeedbackrequest schema
- New schemas for attributes, attribute options, externalFeedbackMessageAssociations
- New schemas for createExternalUserRequest, externalUser, userSegment
- /feedback only accepts an api key as a query, so there’s no global definition for security and instead they’re on each endpoint (should /feedback accept it in the header, too?)
- Added details to clarify that feedback always goes in messages with details on externalMessageId and ProjectID.
- Added links to the repos for the Python and Typescript SDKs

Note:
* The typescript SDK is missing the ability to create users

gmbrown commented:
Added a method to create/update users to the typescript sdk https://github.com/Melodi-fyi/melodi-sdk-typescript/pull/23

gmbrown commented on openapi.yaml:
```diff
@@ -373,62 +465,122 @@ components:
-          type: string
+          nullable: true
+          description: Text content of the feedback
         userId:
```
This is an outdated field and not actually used anymore. I just [updated the api to no longer return it](https://github.com/Melodi-fyi/llm-ops-vercel/pull/1105), so let's remvoe it from here please.

gmbrown commented on api-reference/create-user.mdx:
```diff
@@ -1,7 +1,5 @@
 ---
 title: 'Create User'
```
I think everything in here should say Create or Update User

gmbrown commented on openapi.yaml:
```diff
@@ -373,62 +465,122 @@ components:
+          format: int32
+          nullable: true
+          description: ID of the internal user if applicable
+        externalUserId:
```
I've also updated this to now include the ExternalUser object in the return object but not the id. An example response now looks like this:

```
{
  "id": 313824,
  "feedbackType": "POSITIVE",
  "feedbackText": "This is great",
  "createdAt": "2024-11-08T22:38:01.147Z",
  "updatedAt": "2024-11-08T22:38:01.147Z",
  "externalUser": {
    "id": 4708,
    "organizationId": 1,
    "externalId": "mintlify-test-1",
    "name": "Mr Bob",
    "email": null,
    "username": null,
    "createdAt": "2024-11-08T19:55:35.728Z",
    "updatedAt": "2024-11-08T22:23:50.350Z",
    "segments": []
  },
  "attributeOptions": [
    {
      "id": 39,
      "attributeId": 22,
      "name": "Low",
      "createdAt": "2024-11-08T22:38:00.991Z",
      "updatedAt": "2024-11-08T22:38:00.991Z",
      "attribute": {
        "id": 22,
        "organizationId": 1,
        "projectId": 12074,
        "name": "Humor Level",
        "createdAt": "2024-11-08T22:38:00.952Z",
        "updatedAt": "2024-11-08T22:38:00.952Z"
      }
    }
  ]
}
```

gmbrown reviewed (approved):
Just a few comments going through and running everything inside Mintlify.

I made a couple changes on the api side to make the return objects simpler.

Ultimately, I would love to get all the input/output types defined in the typescript sdk and pulled into our codebase from there, but that might be a project for another day.

sremsen commented on openapi.yaml:
```diff
@@ -10,8 +10,10 @@ paths:
   /threads:
     get:
       summary: Retrieve all threads
+      # Remove security array here since it's now global
```
ah yes AI artifact

sremsen commented on openapi.yaml:
```diff
@@ -247,12 +256,95 @@ paths:
+      in: header
+      name: API-Key
+      description: Your Melodi API key
+    ApiKeyQuery:
```
Should it be ApiKeyHeaderParam too then?

sremsen commented:
Changelog:

- Added details for createUser for the typescript version of the SDKs.mdx file
- Added additional detail to overview.mdx
- Modified examples of feedback attributes and messages in Typescript SDK to match Python SDK
- Updated the ExternalFeedback schema based on the new example response
    - **Removed Fields**:
        - userId
        - externalUserId
        - externalThreadId
        - messageAssociation
    - **Added Fields**:
        - externalUser (references full ExternalUser schema)
    - **Updated Structure**:
        - attributeOptions now includes nested attribute object
        - Simplified required fields to just id, createdAt, and updatedAt

gmbrown commented on getting-started/quickstart.mdx:
```diff
@@ -2,7 +2,7 @@
 ---
 
-You can get started with Melodi in less than 10 minutes. First, you'll need to send session data to Melodi, which you can do with the following options:
+You can get started with Melodi in less than 30 minutes. First, you'll need to send session data to Melodi, which you can do with the following options:
```
Did you mean to change this back?
